File: guestClient/src/app/card/card.ts
import DateTimeFormat = Intl.DateTimeFormat;
// import ObjectID from "bson-objectid";
// import ObjectID from "bson-objectid";

export class Card{

  public id:string;
  public serialNumber:string;
  public location:any;
  public manufactured:DateTimeFormat;
  public validUntil:DateTimeFormat;
  public ctype:any;

  constructor() {}
}

File: guest-registry/guest-registry-data/guest-registry-data-api/src/main/java/practice/guestregistry/data/api/domain/Card.java
package practice.guestregistry.data.api.domain;

import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;
import java.time.LocalDateTime;

@Data
public class Card {
    private String id;
    @NotEmpty
    private String serialNumber;

    @NotNull
    private Location location;
    @NotNull
    private LocalDateTime manufactured;
    @NotNull
    private LocalDateTime validUntil;
    @NotNull
    private String ctype;
}
